### PR TITLE
Feature/#1814 pagination of mods final

### PR DESCRIPTION
**Pull request by Aksiznarf-Uar**
Created pagination for the mod vault.

### CODE REVIEW

File: src/main/java/com/faforever/client/mod/ModVaultController.java [hunk 1]
```diff
@@ -289,6 +326,12 @@ private void appendSearchResult(List<ModVersion> modVersions, Pane pane) {
           return controller;
         }).collect(Collectors.toList());
 
+    Platform.runLater(() -> {
+      pane.getChildren().clear();
+    });
+
+    JavaFxUtil.assertBackgroundThread();
```
Comment (1-alex98) [inline src/main/java/com/faforever/client/mod/ModVaultController.java:333]: if this is really the background thread why load fxml on the background thread just above :thinking:

Comment (1-alex98) [reply src/main/java/com/faforever/client/mod/ModVaultController.java:333]: Also I guess make assert on method top

**Review by 1-alex98 — CHANGES_REQUESTED** [review]
Unit tests failing

File: src/main/java/com/faforever/client/mod/ModVaultController.java [hunk 2]
```diff
@@ -83,13 +87,13 @@
...
   public SearchController searchController;
-  public Button moreButton;
+  public Pagination paginationControl;
 
   private boolean initialized;
   private ModDetailController modDetailController;
   private final ObjectProperty<ModVaultController.State> state;
-  private int currentPage;
   private Supplier<CompletableFuture<List<ModVersion>>> currentSupplier;
+  public String requestedModType;
```
Comment (axel1200) [inline src/main/java/com/faforever/client/mod/ModVaultController.java:96]: enum?

File: src/main/java/com/faforever/client/mod/ModVaultController.java [hunk 3]
```diff
@@ -132,13 +136,40 @@ public void initialize() {
...
+          enterSearchResultState();
+        })
+    .exceptionally(throwable -> {
+      notificationService.addNotification(new ImmediateErrorNotification(
+          i18n.get("errorTitle"), i18n.get("vault.mods.searchError"),
+          throwable, i18n, reportingService
+      ));
+      enterShowroomState();
+      return null;
+    });
```
Comment (1-alex98) [inline src/main/java/com/faforever/client/mod/ModVaultController.java:164]: ident

File: src/main/java/com/faforever/client/mod/ModVaultController.java [hunk 4]
```diff
@@ -132,13 +136,40 @@ public void initialize() {
...
+    paginationControl.currentPageIndexProperty().addListener(new ChangeListener<Number>() {
+      @Override
+      public void changed(ObservableValue<? extends Number> observable, Number oldValue, Number newValue) {
+        updateModPage();
+      }
+    });
+  }
+
+  private void updateModPage(){
+    loadingLabel.setVisible(true);
```
Comment (1-alex98) [inline src/main/java/com/faforever/client/mod/ModVaultController.java:150]: enterLoadingState()

File: src/main/java/com/faforever/client/mod/ModVaultController.java [hunk 5]
```diff
@@ -132,13 +136,40 @@ public void initialize() {
...
+
+    currentSupplier.get()
+        .thenAccept(mods -> {
+          appendSearchResult(mods, searchResultPane);
```
Comment (1-alex98) [inline src/main/java/com/faforever/client/mod/ModVaultController.java:154]: missleading name

**Review by 1-alex98 — CHANGES_REQUESTED** [review]

**Comment by Sheikah45** [comment]
@axel1200 should this be closed?

**Comment by 1-alex98** [comment]
yes

**Pull request closed without merging**